native/python/pyjp_buffer.cpp

```diff
@@ -0,0 +1,139 @@
+#include "jpype.h"
+#include "pyjp.h"
+#include "jp_buffer.h"
+#include "jp_buffertype.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+static void PyJPBuffer_dealloc(PyJPBuffer *self)
+{
+	JP_PY_TRY("PyJPBuffer_dealloc");
+	delete self->m_Buffer;
```

Comment:
This assumes that JPBuffers are always direct. Sorry I didn't spot this check earlier.

Reply:
You can have a buffer which is not direct in this code, but it won't have a memoryview currently.  I left the stubs in case we wanted to implement indirect buffers as well.   

But I am very reluctant to go down that path.   If we implement both direct and java array backed buffers on this interface, then we will suffer the problem that java array buffers can't be made into read/write buffers because the JNI pinning restrictions.   As it stands you can make a memory buffer for a direct buffer which will fail for an indirect buffer, but if it is indirect you can call the array method which will give you the memory buffer with array rules.

I believe this is the "best" compromise in terms of design.   Arrays have one set of rules and (direct) Buffers have a different set.    Obviously some docs would help.   The rules are pretty obvious at the implementation level given the JNI calls, but may be surprising for those not familiar with JNI.

Reply:
I agree, that we do not want to mix these concepts. Some documentation/examples would be great indeed.